cli/worker.rs

```diff
@@ -479,29 +476,6 @@ impl CliMainWorkerFactory {
     let (main_module, is_main_cjs) = if let Ok(package_ref) =
       NpmPackageReqReference::from_specifier(&main_module)
     {
-      let package_ref = if package_ref.req().version_req.version_text() == "*" {
-        // When using the wildcard version, select the same version used in the
-        // package.json deps in order to prevent adding new dependency version
-        shared
-          .options
-          .maybe_root_package_json_deps
-          .as_ref()
-          .and_then(|deps| {
-            deps
-              .values()
-              .filter_map(|v| v.as_ref().ok())
-              .find(|dep| dep.name == package_ref.req().name)
-              .map(|dep| {
-                NpmPackageReqReference::new(PackageReqReference {
-                  req: dep.clone(),
-                  sub_path: package_ref.sub_path().map(|s| s.to_string()),
-                })
-              })
-          })
-          .unwrap_or(package_ref)
-      } else {
-        package_ref
-      };
```

Comment:
This didn't seem to be used/necessary and the test for it passed without it, so I removed this.